src/run-generator/run-generator.c

```diff
@@ -0,0 +1,155 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "proc-cmdline.h"
+#include "strv.h"
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "specifier.h"
+#include "escape.h"
+#include "fileio.h"
+#include "mkdir.h"
+
+static const char *arg_dest = "/tmp";
+static char **arg_commands = NULL;
+static char *arg_success_action = NULL;
+static char *arg_failure_action = NULL;
+
+static int parse(const char *key, const char *value, void *data) {
+        int r;
+
+        if (streq(key, "systemd.run")) {
+
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                r = strv_extend(&arg_commands, value);
+                if (r < 0)
+                        return log_oom();
+
+        } else if (streq(key, "systemd.run_success_action")) {
```

Comment:
Please use `proc_cmdline_key_streq` here and below. (Above too, for consistency.)